rabbitmq/src/main/java/io/micronaut/rabbitmq/intercept/RabbitMQIntroductionAdvice.java

```diff
@@ -296,8 +298,10 @@ private StaticPublisherState getPublisherState(MethodInvocationContext<?, ?> con
 
             String exchange = client.getValue(String.class).orElse("");
             Optional<AnnotationValue<Binding>> bindingAnn = method.findAnnotation(Binding.class);
+            Optional<AnnotationValue<Mandatory>> mandatoryAnn = method.findAnnotation(Mandatory.class);
 
             Optional<String> routingKey = bindingAnn.flatMap(b -> b.getValue(String.class));
+            Optional<Boolean> mandatory = mandatoryAnn.flatMap(m -> m.getValue(Boolean.class));
```

Comment:
Should we use `AnnotationValue::booleanValue` instead?

Reply:
more efficient would be `method.booleanValue(Mandatory.class)` instead of reading the whole annotation just to get one value.

Reply:
Unfortunately, `booleanValue` does not work as expected when `@Mandatory` is applied at TYPE level.

I tried this:
```java
Optional<Boolean> mandatory = mandatoryAnn.flatMap(AnnotationValueResolver::booleanValue);
```

But then the tests fail because the mandatory flag is not detected anymore on this client:

```groovy
@RabbitClient
@Mandatory
static interface MandatorySender1 {
    @Binding("no-such-queue")
    void send(String message)
}
```

Reply:
do we want to support as a type level in case you want all your @Binding methods to use mandatory?

Can you write a test which verify such as in the following example that `@Mandatory` at the method level level takes precedence?

```
@RabbitClient
@Mandatory
static interface MandatorySender1 {
    @Binding("no-such-queue")
    void send(String message)

   @Mandatory(StringUtils.FALSE)
    @Binding("foobar")
    void sendFoobar(String message)
}```

Reply:
@sdelamo 

> Do we want to support as a type level in case you have an interface with multiple methods annotated with `@Binding` and want all of them to use mandatory?

Yes, that was my intention. Allowing `@Mandatory` at type level saves the user from having to annotate each method individually.

> Can you write a test that verifies, as in the following example, that `@Mandatory` at the method level takes precedence?

Sure thing 👍 https://github.com/micronaut-projects/micronaut-rabbitmq/pull/508/commits/92bc9e621373da59d7f0924c5db9f1ae14d9e537